### PR TITLE
[SPARK-31527][SQL][TESTS][FOLLOWUP] Add a benchmark test for datetime add/subtract interval operations

### PR DESCRIPTION
**Pull request by yaooqinn**
### What changes were proposed in this pull request?
With https://github.com/apache/spark/pull/28310, the operation of date +/- interval(m, d, 0) has been improved a lot.

According to the benchmark results, about 75% time cost is reduced because of no casting date to timestamp back and forth.

In this PR, we add a benchmark for these operations, and timestamp +/- interval operations as accessories.

### Why are the changes needed?

Performance test coverage, since these operations are missing in the DateTimeBenchmark.

### Does this PR introduce any user-facing change?

No, just test

### How was this patch tested?

regenerated benchmark results

### CODE REVIEW

**Comment by yaooqinn** [comment]
cc @cloud-fan @dongjoon-hyun @maropu @HyukjinKwon many thank you for your valuable time.

**Review by maropu — APPROVED** [review]

**Review by MaxGekk — COMMENTED** [review]
wholestage on or off doesn't matter actually. Could you build one table with default settings (wholestage on) like `Conversion from/to external types` in this benchmark.

**Comment by yaooqinn** [comment]
> wholestage on or off doesn't matter actually. Could you build one table with default settings (wholestage on) like `Conversion from/to external types` in this benchmark.

Can you explain the reason why it should follow that specific case but not other common ones？

**Comment by MaxGekk** [comment]
> Can you explain the reason why it should follow that specific case but not other common ones？

because
1. You do benchmarking of particular functions, and it doesn't matter from where they are called - from generated code or not
2. Wholestage on is common, and default case
3. As your benchmark result show there are no diffs between wholestage on/off

**Comment by cloud-fan** [comment]
It's benchmark only so we don't need to wait for jenkins.

Thanks, merging to master/3.0!

File: sql/core/benchmarks/DateTimeBenchmark-jdk11-results.txt [hunk 1]
```diff
@@ -1,435 +1,457 @@
+================================================================================================
+datetime +/- interval
+================================================================================================
+
+Java HotSpot(TM) 64-Bit Server VM 11.0.5+10-LTS on Mac OS X 10.15.4
+Intel(R) Core(TM) i9-9980HK CPU @ 2.40GHz
+datetime +/- interval:                    Best Time(ms)   Avg Time(ms)   Stdev(ms)    Rate(M/s)   Per Row(ns)   Relative
+------------------------------------------------------------------------------------------------------------------------
+date + interval(m)                                  919            933          22          0.0   306237514.3       1.0X
```
Comment (MaxGekk) [inline sql/core/benchmarks/DateTimeBenchmark-jdk11-results.txt:9]: The column `Per Row(ns)` is incorrect obviously. This PR https://github.com/apache/spark/pull/28440 fixes the issue.

**Pull request closed without merging**